src/sea.ts

```diff
@@ -0,0 +1,253 @@
+import path from 'path';
+import os from 'os';
+import fs from 'fs-extra';
+import postject from 'postject';
+
+import { spawnPromise } from './spawn';
+import { log } from './utils/log';
+import { SignToolOptions } from './types';
+
+interface SeaOptions {
+  // Full path to the sea. Needs to end with .exe
+  path: string
+  // Optional: A binary to use. Will use the current executable
+  // (process.execPath) by default.
+  bin?: string;
+  // Sign options
+  windowsSign: SignToolOptions
+}
+
+interface InternalSeaOptions extends Required<SeaOptions> {
+  dir: string
+  filename: string
+}
+
+const FILENAMES = {
+  SEA_CONFIG: 'sea-config.json',
+  SEA_MAIN: 'sea.js',
+  SEA_BLOB: 'sea.blob',
+  SEA_RECEIVER: 'receiver.cjs'
+};
+
+const SEA_MAIN_SCRIPT = `
+const bin = "%PATH_TO_BIN%";
+const script = "%PATH_TO_SCRIPT%";
+const options = JSON.parse(\`
+%WINDOWS_SIGN_OPTIONS%
+\`.trim())
+
+const { spawnSync } = require('child_process');
+
+function main() {
+  console.log("@electron/windows-sign sea");
+  console.log({ bin, script });
+
+  try {
+    const spawn = spawnSync(
+      bin,
+      [ script, JSON.stringify(options), JSON.stringify(process.argv.slice(1)) ],
+      { stdio: ['inherit', 'inherit', 'pipe'] }
+    );
+
+    const stderr = spawn.stderr.toString().trim();
+
+    if (stderr) {
+      throw new Error(stderr);
+    }
+  } catch (error) {
+    process.exitCode = 1;
+    throw new Error(error);
+  }
+}
+
+main();
+`;

```

Comment:
This is a great idea, thank you! I tried to get it to work properly - but it's hard to make prettier & co _really_ understand what's going on, as the scripts here even import the module itself.

I'll try to revisit this later, since I do agree with ya. It's a little annoying that the editor can't tell me about my typos in the string.